catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/util/DigitalSignature.java

```diff
@@ -0,0 +1,129 @@
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.commands.util;
+
+import java.io.BufferedInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.security.InvalidKeyException;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.PrivateKey;
+import java.security.Signature;
+import java.security.SignatureException;
+import java.security.UnrecoverableKeyException;
+import java.security.cert.Certificate;
+import org.apache.commons.io.IOUtils;
+import org.codice.ddf.security.common.Security;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class DigitalSignature {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(DigitalSignature.class);
+
+  private static final int BUFFER_SIZE = 1024;
+
+  private static final int OFFSET = 0;
+
+  private KeyStore keyStore;
+
+  public DigitalSignature() {
+    this.keyStore = Security.getInstance().getSystemKeyStore();
+  }
+
+  public DigitalSignature(KeyStore keyStore) {
+    this.keyStore = keyStore;
+  }
+
+  public byte[] createDigitalSignature(InputStream data, String alias, String password)
+      throws IOException {
+    PrivateKey privateKey = getPrivateKey(alias, password);
+
+    if (privateKey == null) {
+      throw new CatalogCommandRuntimeException("");
+    }
+
+    try (BufferedInputStream bufferedInputStream = new BufferedInputStream(data)) {
+      Signature rsa = Signature.getInstance("SHA256withRSA");
+
+      rsa.initSign(getPrivateKey(alias, password));
+
+      byte[] buffer = new byte[BUFFER_SIZE];
+      int len;
+
+      while ((len = bufferedInputStream.read(buffer)) >= 0) {
+        rsa.update(buffer, OFFSET, len);
+      }
+
+      return rsa.sign();
+    } catch (SignatureException | InvalidKeyException | NoSuchAlgorithmException e) {
+      String message = "An error occurred while signing file";
+      LOGGER.debug(message, e);
+      throw new CatalogCommandRuntimeException(message, e);
+    }
+  }
+
+  public boolean verifyDigitalSignature(
+      InputStream data, InputStream signature, String certificateAlias)
+      throws IOException, CatalogCommandRuntimeException {
+    byte[] sigToVerify = IOUtils.toByteArray(signature);
+
+    Certificate certificate = getCertificate(certificateAlias);
+
+    if (certificate == null) {
+      throw new CatalogCommandRuntimeException("");
+    }
+
+    try (BufferedInputStream bufferedInputStream = new BufferedInputStream(data)) {
+      Signature rsa = Signature.getInstance("SHA256withRSA");
+      rsa.initVerify(certificate);
+
+      byte[] buffer = new byte[BUFFER_SIZE];
+      int len;
```

Comment:
why is this declared outside the while loop?

Reply:
Probably to avoid needless re-instantiation of the variable(s) within the loop.

Reply:
Changed this code up a bit. It should make sense now